README.md

```diff
@@ -445,24 +445,33 @@ We have a small collection of algorithms, one for every day of the month. Scroll
     - Now, implement a hash function that takes a string of variable length and returns an integer with a fixed maximum number of digits. Try to come up with your own method of returning a fixed length hash. The simplest method would be to just use a 3-digit length of the string. 001, 042, 340 for 1, 42 and 340 characters respectively. But that's no fun.
     - **Hint:** Try something with the ASCII values of each word maybe.
 
+
   - **_Optional Problem_**
     - Hash functions may return the same output for various inputs. This is known as collision. There are two main collision resolution techniques:
       - [Separate Chaining](https://youtu.be/_xA8UvfOGgU)
       - [Open Addressing](https://youtu.be/Dk57JonwKNk)
     - Implement a hash table with one of the collision resolution techniques mentioned above.
 
+
   - **Uses**
     - As mentioned above, hash functions are often used in combination with a hash table for rapid data lookup.
     - They are also useful in cryptography. A cryptographic hash function allows one to easily verify that some input data maps to a given hash value, but if the input data is unknown, it is deliberately difficult to reconstruct it (or any equivalent alternatives) by knowing the stored hash value.
     - Cryptographic hash functions also play a crucial role in blockchains.
 
+
   - **Resources**
     - [Hash function – Wikipedia](https://en.wikipedia.org/wiki/Hash_function)
     - [Hash table - Wikipedia](https://en.wikipedia.org/wiki/Hash_table)
     - [An in-depth video about cryptocurrencies, blockchain and hash functions](https://youtu.be/bBC-nXj3Ng4)
 
 
 ### **December 20 - Dijkstra's Shortest Reach**
+  - **Terminologies Involved**
+    - A **graph** is a pictorial representation of a set of objects where some pairs of objects are connected by links.
+    - Each node of the graph is represented as a **vertex**.
+    - An **edge** represents a path between two vertices or a line between two vertices.
+    - A **path** represents a sequence of edges between the two vertices.
```

Comment:
So these particular definitions are recursive/confusing xD

An edge represents a path. A path represents a sequences of edges. So an edge represents a sequence of edges?

Reply:
Those are my AU textbook definitions xD
I'll change em!